dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java

```diff
@@ -223,6 +223,20 @@ public long getWorkflowsLength(long collectionId) {
         return this.currentSession().createNamedQuery("Entry.getWorkflowsLength", Long.class).setParameter("collectionId", collectionId).getSingleResult();
     }
 
+    public List<CollectionEntry> getCollectionNotebooks(long collectionId) {
+        List<CollectionEntry> collectionWorkflows =  getCollectionWorkflows(collectionId);
```

Comment:
This means notebooks are going to be a subset of workflows, which seems wrong. It seems like they should be distinct.

Reply:
A core question here is: "What is a workflow?"  Alas, in our naming scheme, it depends...
In the class hierarchy, `Workflow` is the parent class of `BioWorkflow`, `Notebook`, `AppTool`, and `Service`, and so it's not a concrete workflow in the `BioWorkflow` sense, but an abstract thing, more like an `Entry`-plus.
This produces all sorts of confusion.  We use the term `workflow` to mean "BioWorkflow" in some places, and "Entry-plus" in others. 
IMHO, there should probably be methods to retrieve the collection contents for each of the concrete entry types, and maybe a method that retrieves all entries in the collection.

Reply:
> This means notebooks are going to be a subset of workflows, which seems wrong. It seems like they should be distinct.

Is it though? Isn't the query `from Workflow w, Collection col join col.entries as e where type(w) in (BioWorkflow, AppTool, Notebook )` i.e.

Reply:
> IMHO, there should probably be methods to retrieve the collection contents for each of the concrete entry types, and maybe a method that retrieves all entries in the collection.

Fair

Reply:
> > This means notebooks are going to be a subset of workflows, which seems wrong. It seems like they should be distinct.
> 
> Is it though? Isn't the query `from Workflow w, Collection col join col.entries as e where type(w) in (BioWorkflow, AppTool, Notebook )` i.e.

Looks like you may not have finished your thought, but I meant `getCollectionWorkflows()` returns bioworkflows, app tools, and notebooks, while `getCollectionNotebooks()` returns just notebooks. And then further below we set the lengths based on the length of those two collections, which means notebooks would get double counted.

Steve's suggestion of `getCollectionBioWorkflows()`,  etc. would address this.

Reply:
Added methods for each entryTypes. Thank you for everyone's inputs!